packages/trader/src/Modules/Trading/Containers/purchase.jsx

```diff
@@ -77,12 +79,21 @@ const Purchase = ({
         }
     });
 
+    if (components &&
+        components.length &&
+        components.length === 2 &&
```

Comment:
seems redundant, why have `components.length &&` ? we can just compare if it is equal to 2